extensions/src/ACRE2TS/TsCallbacks_static.cpp

```diff
@@ -34,27 +34,32 @@ void ts3plugin_infoData(uint64 serverConnectionHandlerID, uint64 id, enum Plugin
 
     BOOL noAcre = FALSE;
     char *metaData;
+    std::string result;
+    std::string sharedMsg;
 
     switch(type) {
         case PLUGIN_SERVER:
             break;
         case PLUGIN_CHANNEL:
             break;
         case PLUGIN_CLIENT:
-            if (ts3Functions.getClientVariableAsString(serverConnectionHandlerID, (anyID)id, CLIENT_META_DATA, &metaData) == ERROR_ok) {
-                if (!metaData) { 
-                    noAcre = TRUE;
-                }
-                int metaDataLength = strlen(metaData);
-                if (metaDataLength < 3 || metaDataLength > (INFODATA_BUFSIZE - 2)) {
+            if (ts3Functions.getClientVariableAsString(ts3Functions.getCurrentServerConnectionHandlerID(), (anyID)id, CLIENT_META_DATA, &metaData) == ERROR_ok) {
+                if (!metaData) {
                     noAcre = TRUE;
+                } else {
+                    sharedMsg = metaData;
+                    if ((sharedMsg.find(START_DATA) == std::string::npos) || (sharedMsg.find(END_DATA) == std::string::npos)) {
+                        noAcre = TRUE;
+                    } else {
+                        result = sharedMsg.substr(sharedMsg.find(START_DATA) + strlen(START_DATA), sharedMsg.find(END_DATA) - sharedMsg.find(START_DATA) - strlen(START_DATA));
```

Comment:
you are potentially calling `strlen` at runtime. If you have c++17 you can use a string_view's length() method to get the length at compiletime.

Reply:
Also, technically the `.find(END_DATA)` should only start searching AFTER the START_DATA, if someone is funny and puts a END_DATA before a START_DATA in his metadata, ACRE will probably kill itself.

Also you can store `sharedMsg.find(START_DATA)` in a variable instead of calculating it 3 times.
`sharedMsg.find(END_DATA)` too

Reply:
Also if you have c++17 you should do the find/substr's on a string_view to prevent unnecessary copies.

Reply:
Did I make the same mistakes? *shakes head*

Reply:
We can use C++17 no problem.

Reply:
Then C++17 forward on... I was under the impression we had support for only c++14.

Reply:
Since recently `make.py` takes most up-to-date compiler version on the given platform. Jenkins is running VS2019.